browser/ui/webui/brave_webcompat_reporter_ui.cc

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/brave_webcompat_reporter_ui.h"
+
+#include "brave/browser/ui/webui/basic_ui.h"
+#include "brave/common/webui_url_constants.h"
+#include "brave/components/brave_webcompat_reporter/resources/grit/brave_webcompat_reporter_generated_map.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/grit/brave_components_resources.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "content/public/browser/web_ui_message_handler.h"
+
+#include "brave/browser/brave_webcompat_reporter/webcompat_reporter_service.h"
+
+namespace {
+
+class WebcompatReporterDOMHandler : public content::WebUIMessageHandler {
+ public:
+  WebcompatReporterDOMHandler();
+  ~WebcompatReporterDOMHandler() override;
+
+  // WebUIMessageHandler implementation.
+  void RegisterMessages() override;
+
+ private:
+  void HandleSubmitReport(const base::ListValue* args);
+  std::unique_ptr<brave::BraveWebcompatReporterService> sender_;
+
+  DISALLOW_COPY_AND_ASSIGN(WebcompatReporterDOMHandler);
+};
+
+WebcompatReporterDOMHandler::WebcompatReporterDOMHandler()
+    : sender_(std::make_unique<brave::BraveWebcompatReporterService>()) {}
+
+WebcompatReporterDOMHandler::~WebcompatReporterDOMHandler() {}
+
+void WebcompatReporterDOMHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      "brave_webcompat_reporter.submitReport",
+      base::BindRepeating(&WebcompatReporterDOMHandler::HandleSubmitReport,
+                          base::Unretained(this)));
+}
+
+void WebcompatReporterDOMHandler::HandleSubmitReport(const base::ListValue* args) {
+  DCHECK_EQ(args->GetSize(), 1U);
+  std::string site_url;
+  if (!args->GetString(0, &site_url))
+    return;
+
+  sender_->SubmitReport(site_url);
+}
+
+}  // namespace
+
+BraveWebcompatReporterUI::BraveWebcompatReporterUI(content::WebUI* web_ui, const std::string& name)
```

Comment:
we have inconsistent usage of `BraveWebcompat` vs `Webcompat` throughout the code. I think we can drop `Brave` everywhere to make it consistent